packages/vite/src/node/plugins/html.ts

```diff
@@ -777,6 +786,49 @@ export type IndexHtmlTransform =
       transform: IndexHtmlTransformHook
     }
 
+export function preImportMapHook(
+  config: ResolvedConfig
+): IndexHtmlTransformHook {
+  return (html, ctx) => {
+    const importMapIndex = html.match(importMapRE)?.index
+    if (!importMapIndex) return
+
+    const moduleScriptIndex = html.match(moduleScriptRE)?.index
+    if (!moduleScriptIndex) return
+
+    if (moduleScriptIndex < importMapIndex) {
+      const relativeHtml = path.relative(config.root, ctx.filename)
```

Comment:
```suggestion
      const relativeHtml = normalizePath(path.relative(config.root, ctx.filename))
```
I think `normalizePath` is needed here for windows.